agent-inject/agent/config.go

```diff
@@ -94,6 +94,18 @@ func (a *Agent) newTemplateConfigs() []*Template {
 }
 
 func (a *Agent) newConfig(init bool) ([]byte, error) {
+	config_map := make(map[string]interface{})
+	switch a.AutoAuthMethod {
+	case "approle":
+		config_map["role_id_file_path"] = "/vault/approle/roleid"
+		config_map["secret_id_file_path"] = "/vault/approle/secretid"
+		config_map["remove_secret_id_file_after_reading"] = false
+	case "kubernetes":
```

Comment:
This could be replaced by the `GetAutoAuthConfig`

Have two structs implement this interface `KubernetesAutoAuth` and `AppRoleAutoAuth`.